lib/hexpm/accounts/pwned/hexpm.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Hexpm.Accounts.Pwned.Hexpm do
+  @behaviour Hexpm.Accounts.Pwned
+
+  @base_url "https://api.pwnedpasswords.com/"
+  @weakness_threshold 1
+  @timeout 500
+
+  @spec password_breached?(String.t()) :: boolean
+  def password_breached?(string_password) do
+    string_password
+    |> hash_password()
+    |> occurrences_of_hash()
+    |> Kernel.>=(@weakness_threshold)
+  end
+
+  defp hash_password(string_password) do
+    :sha
+    |> :crypto.hash(string_password)
+    |> Base.encode16()
+    |> String.upcase()
+  end
+
+  defp range(searchable_range) do
+    url = @base_url <> "range/#{searchable_range}"
+    headers = [{"User-Agent", "hexpm"}]
+
+    case :hackney.get(url, headers, "", connect_timeout: @timeout, recv_timeout: @timeout) do
+      {:ok, 200, _headers, ref} ->
+        {:ok, body} = :hackney.body(ref)
+        String.split(body, "\r\n")
+
+      {:error, _} ->
+        []
+    end
+  end
+
+  defp occurrences_of_hash(<<searchable_range::bytes-5, remainder::binary>>) do
+    searchable_range
+    |> range()
+    |> Enum.find(fn data ->
+      [hash | _t] = String.split(data, ":")
+      hash == remainder
+    end)
+    |> number_of_occurrences()
```

Comment:
Since the same string is split here and couple lines below, would it make sense to "pre-split" it? I think it could lead to some cleaner code:
```
searchable_range
|> range()
|> Enum.map(&String.split(&1, ":"))
|> Enum.find_value("0", fn
  [^remainder, occurrences] -> occurrences
  _ -> nil
end)
|> try_parse_or_zero()
```
Also - is it possible the API responds with something that is not an integer? If not, `try_parse_or_zero` could be replaced with a simpler `String.to_integer`.

Reply:
the API documentation says it will always return an int after the colon.
On the other hand, there are cases when no matches are found and the response contains to colons at all. 
The reason I ended up with the [h | t] pattern match was to accommodate those cases.
(the one i saw was a string like "The range provided was invalid")

Reply:
The code I proposed should work well with those cases, since we just ignore any lists other than two elements there.

Reply:
Nice! Thanks for the suggestion.